### ISSUE TITLE
BUG RuntimeError: [address=0.0.0.0:33341, pid=464586] The expanded size of the tensor (913) must match the existing size (514) at non-singleton dimension 1.  Target sizes: [3, 913].  Tensor sizes: [1, 514]

### ISSUE THREAD

**Issue by wencan** [xorbitsai/inference#1639]
OS: Fedora Linux 40 (Workstation Edition) x86_64 
Host: 21D0 ThinkBook 14 G4+ ARA 
Kernel: 6.8.11-300.fc40.x86_64 
Uptime: 1 day, 6 hours, 4 mins 
Packages: 2665 (rpm), 42 (flatpak) 
Shell: bash 5.2.26 
Resolution: 2880x1800 
DE: GNOME 46.2 
WM: Mutter 
WM Theme: Adwaita 
Theme: Adwaita [GTK2/3] 
Icons: Adwaita [GTK2/3] 
Terminal: gnome-terminal 
CPU: AMD Ryzen 5 6600H with Radeon Graphics (12) @ 4.564GHz 
GPU: AMD ATI Radeon 680M 
Memory: 8978MiB / 13649MiB 

Python 3.10.14
torch                             2.3.0
xinference                        0.12.0

rag框架为dify 0.6.10
嵌入模型和rerank模型为xinference运行的bce-embedding-base_v1 和 bce-reranker-base_v1
llm为 百度的ERNIE-Speed-128K api
搜索模式为Hybrid Search

试了拿之前创建的知识库，是没问题的，对话中可以正常命中。

重现步骤：
dify创建知识库，导入文本文件
chatbot提问，或者知识库中测试

```
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.
2024-06-14 12:23:42,304 xinference.api.restful_api 463236 ERROR    [address=0.0.0.0:33341, pid=464586] The expanded size of the tensor (913) must match the existing size (514) at non-singleton dimension 1.  Target sizes: [3, 913].  Tensor sizes: [1, 514]
Traceback (most recent call last):
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/api/restful_api.py", line 1089, in rerank
    scores = await model.rerank(
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xoscar/backends/context.py", line 227, in send
    return self._process_result_message(result)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xoscar/backends/pool.py", line 659, in send
    result = await self._run_coro(message.message_id, coro)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xoscar/backends/pool.py", line 370, in _run_coro
    return await coro
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/core/utils.py", line 45, in wrapped
    ret = await func(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/core/model.py", line 87, in wrapped_func
    ret = await fn(self, *args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/core/model.py", line 535, in rerank
    return await self._call_wrapper(
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/core/model.py", line 111, in _async_wrapper
    return await fn(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/core/model.py", line 375, in _call_wrapper
    ret = await asyncio.to_thread(fn, *args, **kwargs)
  File "/usr/lib64/python3.10/asyncio/threads.py", line 25, in to_thread
    return await loop.run_in_executor(None, func_call)
  File "/usr/lib64/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/xinference/model/rerank/core.py", line 202, in rerank
    similarity_scores = self._model.predict(sentence_combinations)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/sentence_transformers/cross_encoder/CrossEncoder.py", line 378, in predict
    model_predictions = self.model(**features, return_dict=True)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/transformers/models/xlm_roberta/modeling_xlm_roberta.py", line 1205, in forward
    outputs = self.roberta(
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/torch/nn/modules/module.py", line 1532, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/torch/nn/modules/module.py", line 1541, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/wencan/Projects/venv/xinference/lib64/python3.10/site-packages/transformers/models/xlm_roberta/modeling_xlm_roberta.py", line 800, in forward
    buffered_token_type_ids_expanded = buffered_token_type_ids.expand(batch_size, seq_length)
RuntimeError: [address=0.0.0.0:33341, pid=464586] The expanded size of the tensor (913) must match the existing size (514) at non-singleton dimension 1.  Target sizes: [3, 913].  Tensor sizes: [1, 514]


```

**Comment by musicthebee** [comment]
i have the same issue. I had tried setting the max_tokens but the issue persists. Any suggestions? Is this a bug or a config issue with models?

**Comment by wencan** [comment]
@musicthebee It should be related to the token length. I switched to an embedding model and rerank model with a maximum token length of 8192, and changed the chunk_size to 8192. This error did not reoccur.

**Comment by musicthebee** [comment]
Here is what I did. I had used "BAAI--bge-large-en-v1.5" model to embedd close to 100K documents and am using "BAAI--bge-reranker-base" for reranking. The error happens only with retrieving certain documents and not all. Looks like some setting during embedding or reranking is the problem in xinference. Switching the embedding model would take a lot of time and is something I would like to avoid. Even then, I am not sure if it will not happen again. Can someone help with why this is happening? Thanks a lot
